compiler/AST/foralls.cpp

```diff
@@ -965,6 +972,12 @@ CallExpr* resolveForallHeader(ForallStmt* pfs, SymExpr* origSE)
 
   FnSymbol* origIterFn = iterCall->resolvedFunction();
 
+  if (!tryFailure && origTarget) {
+    IteratorGroup* igroup = getIteratorGroup(origTarget);
+    if (gotSA) INT_ASSERT(origIterFn == igroup->standalone);
+    else       INT_ASSERT(origIterFn == igroup->leader);
+  }
+
```

Comment:
Where's the part where we actually use the iteratorGroup instead of the tryToken?